### ISSUE TITLE
Manual Item-Level Coloring

### ISSUE THREAD

**Issue by ZINK-ZINK** [AdiAddons/AdiBags#670]
I thought this used to be an option, but I literally spent 30minutes searching through all the options windows and couldn't find it. I want to be able to manually set the MAX Item Level and MIN Item Level Colors then have Adibags automatically gradient through the colors mostly the same way it does now. 

Like ATM my Main is 297 ilvl and I'm a hardcore Pusher(trying to reach +30s Keys). So personally I would set 311 Gear to "Legendary Orange", ~295 Gear to "Epic Purple", ~285 Gear to "Blue", ~270 to Green, etc. But I want to manually adjust those numbers myself each Tier (no offense, but you're just too slow with this for my taste and I have different taste in aesthetic). 

Having 298 Gear show as white item level annoys me, let me just input the range myself so we aren't reliant on addon updates each tier. IIRC that's pretty much the only time sensitive feature of adibags too, so adding this feature would let you just plop a TOC update into curseforge each patch and keep the clueless-masses off your back.

**Comment by Cidan** [comment]
> set the MAX Item Level and MIN Item Level Colors

Just so we're clear, you mean the ilvl numbers that show up over the items, is that correct?

**Comment by ZINK-ZINK** [comment]
> > set the MAX Item Level and MIN Item Level Colors
> 
> Just so we're clear, you mean the ilvl numbers that show up over the items, is that correct?

Yes, the numbers I circled in Pink. See how all the current patch gear is White text? That drives me nuts. I swear I remember there being an option to manually set the item level range ever since BFA, but I couldn't find it anywhere. 

The specific colors don't matter to me all that much, though I'd prefer "Blizzard" style colors(orange/purple/blue/green/white/grey). The really important part to me is that when it is week 1 of a new Tier I want to manually change the new "Max item level color" to the item level of max mythic raid gear, instead of having to wait for yall to update adibags
![AdiBags ILVL](https://user-images.githubusercontent.com/96964163/187013344-ded958fd-3409-43cc-97cc-b7a48f7c4e44.png)
.

**Comment by Cidan** [comment]
Let me think about the best way to go about this. I have concerns with offering range customization here over an extremely wide (>100,000 users) audience. I think the more conservative first step will be to keep ranges up-to-date, while allowing users to customize the color for each range.

**Comment by ZINK-ZINK** [comment]
You already have an, IMO, equally as "risky" feature implemented though in: [Hide Item Levels]. My request for "Blizzard style colors" is an interconnected, but also entirely separate issue. 

An, IMO, reasonable compromise would be to use the same interface for [Hide Item Levels under xxx range] and just add another edit-box that lets the user input what item level they consider to be Max(BIS ilvl) and Min(low level junk I'd never wear). 

I really don't see what the user could break with that setting, the worst I can imagine is confusing rainbow colored item levels, sounds kinda funny and cute lol. Are you worried users will accidentally vendor good gear because the colors confused them? Cause I LITERALLY have that problem right now. I've already vendored and had to buy-back I think 3 piece of 290+ item level gear because the text was White and I thought it was garbage. 

Hell, you could even link the Min/Max item level colors directly to [Hide Items below xxx ilvl]. 

IRL happens. You might have more time to update adibags more quickly for the next Tier or 2, but what happens when you go on vacation or quit WoW and can't update adibags quickly? Then hardcore raiders are back to having their best gear look like vendor trash.

**Comment by Cidan** [comment]
AdiBags' philosophy is to offer a "hands off" approach to bag sorting -- in many ways it has succeeded in this, and in some ways it has aged out. I need to be clear that AdiBags does not optimize for the power user such as yourself. You're in the top maybe 1-5% of users that play WoW by the numbers, taking into account survivor-ship bias.

That being said, my concern is an overwhelming amount of options for less experienced users can, and will, lead to a degraded experience. For example, should users forget to update their manual numbering setup, or mess with the setting and not understand exactly what it does, this will lead to a degraded experience and possibly attrition. As terrible as it sounds, I would rather lose the top 1-5% than the other 95%.

We're going to stick with a strongly opinionated experience on this for now, but we'll add color customization as a half-way point. We may evaluate this in the future, but not until other design considerations have been taken into account.

Thanks.

**Comment by ZINK-ZINK** [comment]
Fair enough. I don't like it, but I can't really argue with the logic. I stopped using Pitbull3/Pitbull4 long ago precisely because it had become such a massively deep rabbit hole of options. I use SuF and Grid now, but even Grid's Menus are starting to look an awful lot like Vuhdo. 

Okay maybe how about this: In the LUA files themselves(like when I open them in Notepad++): Could you comment in some short explanations where some of the "cVars" are? Or maybe add a few specific well-documented editable variables at the top of the LUA for people like me? That way the 90% never even see the option, but users who can self-diagnose could still fix QoL issues without having to try and understand your entire code?

Like the lines I see in some addon's LUA that are like:

*******************************************************
CHANGE AT YOUR OWN RISK
MinItemLevel=262
MaxItemLevel=311
*******************************************************
DO NOT EDIT BELOW THIS LINE
*******************************************************
codecodecode codecode 
codecode; code

**Comment by Talyrius** [comment]
https://github.com/AdiAddons/AdiBags/blob/df5f499bc264a0e64da1189f885d57b01774e867/modules/ItemLevel.lua#L429-L432
